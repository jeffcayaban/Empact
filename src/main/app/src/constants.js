export const API_BASE_URL = 'http://localhost:5000/api';
export const ACCESS_TOKEN = 'accessToken';
export const CURRENT_USERNAME = 'username';

export const AUTHORIZATION_HEADER_FIELD = "Authorization";
export const AUTHORIZATION_TOKEN_TYPE = "Bearer";

// Website Name
export const WEBSITE_NAME = "Empact";

// User Roles
export const ROLE_USER = "ROLE_USER";
export const ROLE_ADMIN = "ROLE_ADMIN";

// Argument Types
export const EXPERT_OPINION = "ExpertOpinion";
export const POPULAR_OPINION = "PopularOpinion";

// Argumentation Schemes
export const ARGUMENTATION_SCHEMES = [
    'By Expert Opinion',
    'By Popular Opinion'
];

// Critical Question IDs
export const CRITICAL_QUESTION_IDS = [
    'CQ1',	// Are the circumstances true?
    'CQ2',
    'EO_CQ1',
    'PO_CQ1'
];

// Content Types
export const USER = "User";
export const PETITION = "Petition";
export const ARGUMENT = "Argument";

// Content Properties
export const USERNAME = "Username";
export const ID = "Id";

// Content Actions
export const DELETE = "Delete";
export const CLOSE = "Close";

// Sort Types
export const DESCENDING = "desc";
export const ASCENDING = "asc";
export const SORT_BY_VALUES = [DESCENDING, ASCENDING];

// Content Sort By Values
export const NEWEST = "Newest";
export const OLDEST = "Oldest";
export const MOST_DISCUSSED = "Most Discussed";
export const LEAST_DISCUSSED = "Least Discussed";
export const DISPLAY_SORT_BY_VALUES = [NEWEST, OLDEST, MOST_DISCUSSED, LEAST_DISCUSSED];

// Used in Homepage
export const NO_OF_MOST_DISCUSSED_PETITIONS = 3;
export const NO_OF_MOST_DISCUSSED_ARGUMENTS = 3;

// Petition Form Fields
export const PETITION_TITLE_ID = "title";
export const PETITION_SITUATION_ID = "situation";
export const PETITION_ACTION_ID = "action";
export const PETITION_GOAL_ID = "goal";
export const PETITION_VALUE_ID = "value";
export const PETITION_CLOSING_DATE_ID = "closingDateTime";
export const PETITION_IS_ANONYMOUS_ID = "isAnonymous";

// Register User Fields
export const USERNAME_ID = "username";
export const FIRST_NAME_ID = "firstName";
export const LAST_NAME_ID = "lastName";
export const PASSWORD_ID = "password";
export const CONFIRM_PASSWORD_ID = "confirm-password";

// Register User Field Validation
export const PASSWORD_LENGTH_MIN = 5;
export const PASSWORD_LENGTH_MAX = 21;
export const LAST_NAME_LENGTH_MAX = 40;
export const FIRST_NAME_LENGTH_MAX = 40;
export const USERNAME_LENGTH_MAX = 15;
export const MAXIMUM_NO_OF_SOURCES = 2;
export const MINIMUM_NO_OF_SOURCES = 0;

// Petition Form Validation
export const TITLE_LENGTH_MAX = 90;
export const PETITION_FIELD_MAX = 3000;

// Argument Form Validation
export const EXPERT_FIELD_MAX = 90;
export const ARGUMENT_FIELD_MAX = 6000;

// For Argument Data Visualisation
export const TREE_NETWORK_HEIGHT = 400;

// Other
export const VALID_URL_PREFIX_REGEX = /^(https:\/\/|http:\/\/).*/;
export const PETITION_LIST_SIZE = 5;
export const NOTIFICATION_AUTO_CLOSE_MS = 5000;
export const PETITION_ENTRY_SITUATION_MAX_LENGTH = 400;
export const PETITION_ENTRY_SITUATION_ALT_MAX_LENGTH = 200;
export const MOBILE_MAX_WIDTH = 768;