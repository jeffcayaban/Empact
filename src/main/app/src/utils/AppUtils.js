import {API_BASE_URL, AUTHORIZATION_HEADER_FIELD} from '../constants';
import {ACCESS_TOKEN, AUTHORIZATION_TOKEN_TYPE, PETITION_LIST_SIZE} from "../constants";

/** Creates a request object and sends the request to the server. **/
const request = (options) => {
    const headers = new Headers({ 'Content-Type': 'application/json' });
    const accessToken = localStorage.getItem(ACCESS_TOKEN);

    if (accessToken !== undefined && accessToken !== null) {
        headers.append(AUTHORIZATION_HEADER_FIELD, `${AUTHORIZATION_TOKEN_TYPE} ${accessToken}`)
    }

    options = {...options, headers: headers};

    return fetch(options.url, options).then(response =>
            (response.json()).then(jsonResponse => (!response.ok) ? Promise.reject(jsonResponse) : jsonResponse)
    );
};

// ---------------------------- APIs ----------------------------

/** Gets all the petitions **/
export function getAllPetitions(page, size, sort, showClosed) {
    page = page || 0;
    size = size || PETITION_LIST_SIZE;

    return request({
        url: API_BASE_URL + "/petitions?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Gets a single petition given an ID **/
export function getPetition(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/" + petitionId,
        method: 'GET'
    });
}

/** Creates a new petition given its details **/
export function createPetition(petitionData) {
    return request({
        url: API_BASE_URL + "/petitions/create",
        method: 'POST',
        body: JSON.stringify(petitionData)
    });
}

/** Deletes an existing petition given its details **/
export function deletePetition(deletePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/delete",
        method: 'DELETE',
        body: JSON.stringify(deletePetitionData)
    });
}

/** Updates an existing petition with the given details **/
export function editPetition(editPetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/edit/",
        method: 'POST',
        body: JSON.stringify(editPetitionData)
    });
}

/** Logs in into a user account **/
export function login(loginRequest) {
    return request({
        url: API_BASE_URL + "/auth/signInUser",
        method: 'POST',
        body: JSON.stringify(loginRequest)
    });
}

/** Creates a new user account given the account details **/
export function signup(signupRequest) {
    return request({
        url: API_BASE_URL + "/auth/signUpUser",
        method: 'POST',
        body: JSON.stringify(signupRequest)
    });
}

/** Checks the username's availability given a username **/
export function checkUsernameAvailability(username) {
    return request({
        url: API_BASE_URL + "/user/checkUsernameAvailability?username=" + username,
        method: 'GET'
    });
}

/** Deletes a user **/
export function deleteUser(deleteUserRequest) {
    return request({
        url: API_BASE_URL + "/auth/deleteUser",
        method: 'DELETE',
        body: JSON.stringify(deleteUserRequest)
    })
}

/** Fetches the current user **/
export function getCurrentUser() {
    if(!localStorage.getItem(ACCESS_TOKEN)) {
        return Promise.reject("No access token set.");
    }

    return request({
        url: API_BASE_URL + "/user/getCurrentUser",
        method: 'GET'
    });
}

/** Fetches a user profile given a user's username **/
export function getUserProfile(username) {
    return request({
        url: API_BASE_URL + "/user/getUserProfile?username=" + username,
        method: 'GET'
    });
}

/** Fetches the petitions for a given user **/
export function getUserPetitions(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/user?username=" + username + "&page=" + page + "&size=" + size + "&sort=" + sort
        + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Fetches the most discussed petitions for a given user **/
export function getMostDiscussedPetitionsByUser(page, size, sort, username, includeAnon, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitionsByUser?username=" + username + "&page=" + page +
            "&size=" + size + "&sort=" + sort
            + "&includeAnon=" + includeAnon + "&showClosed=" + showClosed,
        method: 'GET',
    });
}

/** Closes a petition given its details **/
export function closePetition(closePetitionData) {
    return request({
        url: API_BASE_URL + "/petitions/close",
        method: 'POST',
        body: JSON.stringify(closePetitionData)
    })
}

/** Creates a new argument by expert opinion given the new details **/
export function createArgumentByExpertOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Creates a new argument by popular opinion given the new details **/
export function createArgumentByPopularOpinion(newArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/createArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(newArgumentData)
    });
}

/** Fetches the arguments for a given petition **/
export function getArgumentsByRootPetitionId(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByRootPetitionId?rootPetitionId=" + petitionId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the arguments for a given argument **/
export function getArgumentsByArgumentId(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByArgumentId?argumentId=" + argumentId + "&isSupporting=" +
            isSupporting + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET',
    });
}

/** Fetches the argument for a given ID **/
export function getArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/" + argumentId,
        method: 'GET'
    });
}

/** Fetches the parent argument for a given argument **/
export function getParentArgumentById(argumentId) {
    return request({
        url: API_BASE_URL + "/arguments/getParentArgumentById?argumentId=" + argumentId,
        method: 'GET',
    });
}

/** Checks whether a petition is closed or not **/
export function isPetitionClosed(petitionId) {
    return request({
        url: API_BASE_URL + "/petitions/isPetitionClosed/" + petitionId,
        method: 'GET'
    })
}

/** Fetches the arguments for a given user **/
export function getUserArguments(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getArgumentsByUsernameAndAnonymity?username=" + username +
            "&showAnonymousContent=" + showAnonymousContent + "&page=" + page + "&size=" + size + "&sort=" + sort,
        method: 'GET'
    })
}

/** Fetches the number of supporting or opposing arguments for a given petition **/
export function countArgsByPetitionIdAndIsSupporting(petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countArgsByPetitionIdAndIsSupporting?petitionId=" + petitionId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Fetches the number of supporting or opposing arguments for a given argument **/
export function countArgsByArgumentIdAndIsSupporting(argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/countAllByArgumentIdAndIsSupporting?argumentId=" + argumentId
            + "&isSupporting=" + isSupporting,
        method: 'GET',
    })
}

/** Deletes an argument given its details **/
export function deleteArgument(deleteArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/delete",
        method: 'DELETE',
        body: JSON.stringify(deleteArgumentData)
    });
}

/** Updates an existing argument by popular opinion with new details **/
export function editArgumentByPopularOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByPopularOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Updates an existing argument by expert opinion with new details **/
export function editArgumentByExpertOpinion(editArgumentData) {
    return request({
        url: API_BASE_URL + "/arguments/editArgumentByExpertOpinion",
        method: 'POST',
        body: JSON.stringify(editArgumentData)
    });
}

/** Fetches the most discussed petitions **/
export function getMostDiscussedPetitions(page, size, sort, showClosed) {
    return request({
        url: API_BASE_URL + "/petitions/getMostDiscussedPetitions?page=" + page + "&size=" + size + "&sort=" + sort +
            "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments **/
export function getMostDiscussedArguments(pageSize, showClosed) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArguments?size=" + pageSize + "&showClosed=" + showClosed,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given petition **/
export function getMostDiscussedArgumentsForPetition(page, size, sort, petitionId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForPetition?page=" + page + "&size=" + size + "&sort=" + sort +
            "&petitionId=" + petitionId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given argument **/
export function getMostDiscussedArgumentsForArgument(page, size, sort, argumentId, isSupporting) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForArgument?page=" + page + "&size=" + size + "&sort=" + sort +
            "&argumentId=" + argumentId + "&isSupporting=" + isSupporting,
        method: 'GET'
    });
}

/** Fetches the most discussed arguments for a given user **/
export function getMostDiscussedArgumentsForUser(page, size, sort, username, showAnonymousContent) {
    return request({
        url: API_BASE_URL + "/arguments/getMostDiscussedArgsForUser?page=" + page + "&size=" + size + "&sort=" + sort +
            "&username=" + username + "&showAnonymousContent=" + showAnonymousContent,
        method: 'GET'
    });
}

/** Fetches the data nodes required for the data visualisation of arguments **/
export function getDataNodes(contentId, isPetition) {
    return request({
        url: API_BASE_URL + "/arguments/getDataNodes?contentId=" + contentId + "&isPetition=" + isPetition,
        method: 'GET'
    });
}

/** Fetches data required for the petition report **/
export function getPetitionReport(petitionId) {
    return request({
        url: API_BASE_URL + "/arguments/getPetitionReport?petitionId=" + petitionId,
        method: 'GET'
    });
}

/** Searches for petitions given a query **/
export function searchPetitions(page, size, sort, showClosed, query) {
    return request({
        url: API_BASE_URL + "/petitions/search?page=" + page + "&size=" + size + "&sort=" + sort + "&showClosed=" + showClosed
         + "&query=" + query,
        method: 'GET'
    });
}
