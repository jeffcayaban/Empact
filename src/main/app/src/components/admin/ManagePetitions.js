import React from 'react';
import {
    generateAdminActionSuccessDialog,
    generateConfirmActionDialog,
    generateSomethingWentWrongDialog,
    isUserAdmin
} from "../../utils/Helpers";
import NotPermittedAlert from "../common/alerts/NotPermittedAlert";
import { closePetition, deletePetition } from "../../utils/AppUtils";
import ManageContentPanel from "./ManageContentPanel";
import { CLOSE, DELETE, PETITION } from "../../constants";

/**
 * Provides the administrator with links to different pages that focuses on managing the petitions within the system.
 */

class ManagePetitions extends React.PureComponent {

    // Makes the request to the server to delete a petition.
    handleDeletePetition(result, petitionId) {
        if (result.value) {
            // Assumes that the user has confirmed the action to delete the petition.
            deletePetition({ petitionId: petitionId })
                .then(() => generateAdminActionSuccessDialog(PETITION, DELETE))
                .catch(() => generateSomethingWentWrongDialog())
        }
    }

    // Confirms the action to delete a petition.
    handleDeletePetitionRequest(e, petitionId) {
        e.preventDefault();
        const confirmActionDialog = generateConfirmActionDialog(PETITION, DELETE);
        confirmActionDialog.then((result) => this.handleDeletePetition(result, petitionId));
    }

    // Makes the request to the server to close a petition.
    handleClosingPetition(result, petitionId) {
        if (result.value) {
            // Assumes that the user has confirmed the action to close the petition.
            closePetition({ petitionId: petitionId })
                .then(() => generateAdminActionSuccessDialog(PETITION, CLOSE))
                .catch(() => generateSomethingWentWrongDialog())
        }
    }

    // Confirms the action to close a petition.
    handleClosingPetitionRequest(e, petitionId) {
        e.preventDefault();
        const confirmActionDialog = generateConfirmActionDialog(PETITION, CLOSE);
        confirmActionDialog.then((result) => this.handleClosingPetition(result, petitionId))
    }

    render() {
        const { isAuthenticated, currentUser } = this.props;

        if (isUserAdmin(currentUser, isAuthenticated)) {
            // Administrators will be able to access the contents of this component.

            return (
                <div>
                    <div className={"centerStyle"}>
                        <h2>{`Manage ${PETITION}s`}</h2>
                    </div>
                    <div className={"manageContentActionPanels"}>
                        <ManageContentPanel
                            contentType={PETITION}
                            contentAction={DELETE}
                            handleSubmit={this.handleDeletePetitionRequest.bind(this)}
                            {...this.props}
                        />
                        <ManageContentPanel
                            contentType={PETITION}
                            contentAction={CLOSE}
                            handleSubmit={this.handleClosingPetitionRequest.bind(this)}
                            {...this.props}
                        />
                    </div>
                </div>
            );
        } else {
            // Users other than administrators will be shown an alert.

            return <NotPermittedAlert />
        }
    }
}

export default ManagePetitions;