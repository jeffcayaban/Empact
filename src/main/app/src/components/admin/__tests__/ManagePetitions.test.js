import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import { ROLE_ADMIN } from "../../../constants";
import ManagePetitions from "../ManagePetitions";
import { CURRENT_USERNAME } from "../../../constants";
import * as AppUtils from "../../../utils/AppUtils";

describe('ManagePetitions', () => {

    it('should render correctly given the content type is User', () => {
        const currentUser = { authorities: [ROLE_ADMIN] };

        localStorage.setItem(CURRENT_USERNAME, "test");
        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
    });

    it('should render an alert given the user is not logged in.', () => {
        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={null}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should correctly call handleClosingPetitionRequest', () => {
        const testId = 'testId';
        const currentUser = { authorities: [ROLE_ADMIN] };
        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        const instance = component.instance();
        const mockEvent = { preventDefault: jest.fn() };

        instance.handleClosingPetitionRequest(mockEvent, testId);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
    });

    it('should correctly call handleDeletePetitionRequest', () => {
        const testId = 'testId';
        const currentUser = { authorities: [ROLE_ADMIN] };
        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        const instance = component.instance();
        const mockEvent = { preventDefault: jest.fn() };

        instance.handleDeletePetitionRequest(mockEvent, testId);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
    });

    it('should correctly call handleClosingPetition', () => {

        const testId = 'testId';
        const currentUser = { authorities: [ROLE_ADMIN] };
        const originalHandleClosingPetition = AppUtils.closePetition;
        AppUtils.closePetition = () => Promise.resolve();

        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        const instance = component.instance();
        instance.handleClosingPetition({ value: true }, testId);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
        AppUtils.closePetition = originalHandleClosingPetition;
    });

    it('should unsuccessfully call handleClosingPetition given the API call was not successful', () => {
        const testId = 'testId';
        const currentUser = { authorities: [ROLE_ADMIN] };
        const originalHandleClosingPetition = AppUtils.closePetition;
        AppUtils.closePetition = () => Promise.reject();

        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        const instance = component.instance();
        instance.handleClosingPetition({ value: true }, testId);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
        AppUtils.closePetition = originalHandleClosingPetition;
    });

    it('should correctly call handleDeletePetition', () => {

        const testId = 'testId';
        const currentUser = { authorities: [ROLE_ADMIN] };
        const originalHandleClosingPetition = AppUtils.deletePetition;
        AppUtils.deletePetition = () => Promise.resolve();

        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        const instance = component.instance();
        instance.handleDeletePetition({ value: true }, testId);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
        AppUtils.deletePetition = originalHandleClosingPetition;
    });

    it('should unsuccessfully call handleDeletePetition given the API call was not successful', () => {
        const testId = 'testId';
        const currentUser = { authorities: [ROLE_ADMIN] };
        const originalHandleClosingPetition = AppUtils.deletePetition;
        AppUtils.deletePetition = () => Promise.reject();

        localStorage.setItem(CURRENT_USERNAME, "test");

        const component = shallow(<ManagePetitions
            isAuthenticated={true}
            currentUser={currentUser}
        />);

        const instance = component.instance();
        instance.handleDeletePetition({ value: true }, testId);

        expect(toJson(component)).toMatchSnapshot();

        localStorage.removeItem(CURRENT_USERNAME);
        AppUtils.deletePetition = originalHandleClosingPetition;
    });
});
