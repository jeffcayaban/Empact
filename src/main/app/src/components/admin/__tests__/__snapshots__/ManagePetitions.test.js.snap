// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagePetitions should correctly call handleClosingPetition 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;

exports[`ManagePetitions should correctly call handleClosingPetitionRequest 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;

exports[`ManagePetitions should correctly call handleDeletePetition 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;

exports[`ManagePetitions should correctly call handleDeletePetitionRequest 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;

exports[`ManagePetitions should render an alert given the user is not logged in. 1`] = `<NotPermittedAlert />`;

exports[`ManagePetitions should render correctly given the content type is User 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;

exports[`ManagePetitions should unsuccessfully call handleClosingPetition given the API call was not successful 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;

exports[`ManagePetitions should unsuccessfully call handleDeletePetition given the API call was not successful 1`] = `
<div>
  <div
    className="centerStyle"
  >
    <h2>
      Manage Petitions
    </h2>
  </div>
  <div
    className="manageContentActionPanels"
  >
    <ManageContentPanel
      contentAction="Delete"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
    <ManageContentPanel
      contentAction="Close"
      contentType="Petition"
      currentUser={
        Object {
          "authorities": Array [
            "ROLE_ADMIN",
          ],
        }
      }
      handleSubmit={[Function]}
      isAuthenticated={true}
    />
  </div>
</div>
`;
